frontend/system/src/Admin/Configuration/OptionListConfigTab.tsx

```diff
@@ -17,46 +17,93 @@ import {
   useColumns,
   useTableStore,
 } from '@common/ui';
-import { useEditModal } from '@common/hooks';
+import { useEditModal, useNotification } from '@common/hooks';
 import { OptionEditModal } from './OptionEditModal';
-
-// TODO: this type should come from gql codegen types
-type ListOption = {
-  id: string;
-  label: string;
-  value: string;
-};
+import {
+  ConfigurationItemNode,
+  ConfigurationItemTypeInput,
+} from '@common/types';
+import { useConfigurationItems } from './api';
+import { ConfigurationItemFragment } from './api/operations.generated';
+import { useDeleteConfigurationItem } from './api/hooks/useDeleteConfigurationItem';
+import { LocalStorage } from 'frontend/common/src';
 
 type OptionListConfigTabProps = {
-  data: ListOption[];
+  type: ConfigurationItemTypeInput;
   category: string;
 };
 
+type DeleteError = {
+  name: string;
+  message: string;
+};
+
 const OptionListConfigTabComponent = ({
-  data,
+  type,
   category,
 }: OptionListConfigTabProps) => {
   const t = useTranslation('system');
-  const { onOpen, onClose, isOpen, entity, mode } = useEditModal<ListOption>();
+  const { success, info, error } = useNotification();
+  const { mutateAsync: deleteItem } = useDeleteConfigurationItem();
+  const { onOpen, onClose, isOpen, entity, mode } =
+    useEditModal<ConfigurationItemFragment>();
+
+  const { data, isError, isLoading } = useConfigurationItems({
+    type: type,
+  });
 
+  const [deleteErrors, setDeleteErrors] = React.useState<DeleteError[]>([]);
   const selectedRows = useTableStore(state =>
     Object.keys(state.rowState)
       .filter(id => state.rowState[id]?.isSelected)
       .map(selectedId => data?.find(({ id }) => selectedId === id))
       .filter(Boolean)
   );
 
-  const columns = useColumns<ListOption>([
+  const columns = useColumns<ConfigurationItemFragment>([
     {
-      key: 'value',
+      key: 'name',
       label: 'label.value',
     },
     'selection',
   ]);
 
-  const deleteAction = (rows: (ListOption | undefined)[]) => {
-    // TODO: implement delete
-    console.log('Rows to delete: ', rows);
+  const deleteAction = (rows: (ConfigurationItemFragment | undefined)[]) => {
```

Comment:
Just found a flaw in deleting these... while we store the values as strings against the entities, when we go to edit it, the dropdowns use the current config items as their options. See picture... I deleted the Oral option and now the disabled dropdown shows blank:

<img width="802" alt="Screenshot 2024-01-12 at 10 15 40 AM" src="https://github.com/msupply-foundation/unified-codes/assets/55115239/3bdce8fa-2a9d-4ad7-9972-8f01902b131a">

Perhaps in that edit form, we could append `entity.name` to the options list, so at least it displays correctly?

Otherwise I think we would be getting into storing some kind of soft delete flag to determine if its available as a selectable option or just a displayable option, which sounds like more work...

I think we'd run into the same issue if editing config item names was allowed too!

Reply:
[8c4f82d](https://github.com/msupply-foundation/unified-codes/pull/504/commits/8c4f82d1af466c33d804b69c413af4ad92d1c710) I think this solves it quite nicely...